Improwing speed and reduce code size when fast_float is using as internal parser code.

IRainman opened the pull request:
### 🚀 Performance and Code Size Improvements

* Optimized parsing logic for use as an **internal lightweight parser** (for example, inside other libraries).
* Reduced binary size by conditionally excluding unneeded parsing features.
* The `from_chars_result_t` structure is reduced to **4 bytes** for better memory efficiency.
* Improved the `parsed_number_string_t` layout and increased `constexpr`/`consteval` propagation to enable compile-time optimizations.
* Reduced register and cache pressure and branching in parsing hot paths.
* Improved performance in both 64-bit and 32-bit builds for all supported types.

---

### ⚙️ New Configuration Macros

Introduced new optional macros to minimize overhead when certain parsing features are not required:

* **`FASTFLOAT_ONLY_POSITIVE_C_NUMBER_WO_INF_NAN`**
  Restricts parsing to **positive C-style numbers only** — no sign characters, no `INF`, `INFINITY`, or `NaN` and no additional options like skip white spaces or support for Fortran or JSON.

* **`FASTFLOAT_ONLY_ROUNDS_TO_NEAREST_SUPPORTED`**
  Assumes that only the **IEEE 754 “round-to-nearest”** rounding mode is needed, removing crutches support code for other modes.

* **`FASTFLOAT_ISNOT_CHECKED_BOUNDS`**
  Disables bounds checking for input ranges that are assumed to be valid.
  Use only when inputs are guaranteed safe — this reduces branching and slightly improves performance.

* **`FASTFLOAT_ASSUME`**
  Provides a portable abstraction for the compiler’s `[[assume]]` intrinsic.

* **`FASTFLOAT_HAS_BYTESWAP`**
  Uses std::byteswap if available to reduce code size and speed up.

* **`FASTFLOAT_HAS_BIT_CAST`**
  Uses std::bit_cast if available to reduce code size and speed up.

---

### 🧩 Remove Deprecated Macros

* Removed obsolete macroses **`FASTFLOAT_ALLOWS_LEADING_PLUS`** and **`FASTFLOAT_SKIP_WHITE_SPACE`**, which are now superseded by the stricter limited-mode options above.

---

### 🧠 Internal Refactoring and Quality Improvements

* Simplified internal parsing logic with compile-time branching controlled by the new macros.
* Enhanced `constexpr` coverage across the codebase for greater compile-time evaluation.
* Updated benchmarks and tests to validate new restricted configurations.
* Fixed minor compiler warnings and addressed **PVS-Studio** static analysis feedback.
* Properly use FASTFLOAT_SIMD_DISABLE_WARNINGS and FASTFLOAT_SIMD_RESTORE_WARNINGS only for instructions that allow unaligned loads.
* Many small cleanups and fixes.
* Added more comments about functionality and realization details.

---

### 📘 Documentation and Build

* Updated **README.md** to describe all new configuration macros and their intended use cases.
* Adjusted build scripts and benchmark configurations for the new compile-time flags.

---

✅ **Result:**
Smaller, faster by default, and more configurable builds — now library can be used for **internal numeric parsers** or **embedded environments** — while maintaining full compatibility, functionality, and performance improvements in the default configuration.

dalle commented:
I recently changed some compile time config options into runtime config options. I believe that both worlds would be good. I would prefer if the compile time options to not be any ifdef-macros, but controlled by using template arguments.

lemire commented:
In this instance, the argument here is to reduce compiled size and improve runtime performance we avoid checking negatives ('-') and for 'nan'/'inf'. In my view, it is an extraordinary claim that checking for the negative ('-') and for 'nan'/'inf' makes a large difference.

It is possibly, but I'd like to see the numbers. Please note that we have do have benchmarks included in the library,

>  Because current realisation parse options in fully software mode and this is bad.

How did you benchmark to arrive at the conclusion that it is bad?

> I recently changed some compile time config options into runtime config options.

I would not assume that it makes a difference to the performance unless we have a hard data. Some parameters can be passed as template arguments, certainly... 

But we need to take into account that parsing a single number takes hundreds of instructions. Predictable branches are not a big concern.

IRainman commented:
> I recently changed some compile time config options into runtime config options. I believe that both worlds would be good. I would prefer if the compile time options to not be any ifdef-macros, but controlled by using template arguments.

Very well! I'm currently reworking my patch to use templates argument and want to add more constexpr / consteval to the code.

IRainman commented:
> In this instance, the argument here is to reduce compiled size and improve runtime performance we avoid checking negatives ('-') and for 'nan'/'inf'. In my view, it is an extraordinary claim that checking for the negative ('-') and for 'nan'/'inf' makes a large difference.
> 
> It is possibly true, but I'd like to see the numbers. Please note that we have do have benchmarks included in the library.
> 

I'm measuring performance on the mathematical parser, that already processes minus signs, and I yesterday rewrote it to also process inf / infinity for mathematical questions. In mathematical parser nan isn't possible at all because if it's nan the parser stops processing equations immediately. 

> > Because current realisation parse options in fully software mode and this is bad.
> 
> How did you benchmark to arrive at the conclusion that it is bad?
> 
Options it's not fully processed by the template and generate many additional lines of code. Also complete removing the minus sign for mantissa and inf/nan from fast_float significantly reduces generated assembler.

To check the performance of such a case, tests should contain only positive numbers in general notation.

dalle commented:
> > I recently changed some compile time config options into runtime config options. I believe that both worlds would be good. I would prefer if the compile time options to not be any ifdef-macros, but controlled by using template arguments.
> 
> Very well! I'm currently reworking my patch to use templates argument and want to add more constexpr / consteval to the code.

Please don't do this until we have the performance comparisons on the gains.

IRainman commented:
> > > I recently changed some compile time config options into runtime config options. I believe that both worlds would be good. I would prefer if the compile time options to not be any ifdef-macros, but controlled by using template arguments.
> > 
> > 
> > Very well! I'm currently reworking my patch to use templates argument and want to add more constexpr / consteval to the code.
> 
> Please don't do this until we have the performance comparisons on the gains.

Of course, I just cleaned my code of defines and replaced them with an option. Unfortunately, code isn't fully clean-up at compile time as a preprocessing macro.

IRainman commented:
P. S. my current version of the test code is: 
`	[[assume(_view._Unchecked_end() - _view._Unchecked_begin() >= 1)]];
	double val;
	constexpr auto options{fast_float::chars_format::general | fast_float::chars_format::no_infnan | fast_float::chars_format::disallow_leading_sign};
	const auto res = fast_float::from_chars(_view._Unchecked_begin(), _view._Unchecked_end(), val, options);
	[[assume(res.ptr - _view._Unchecked_begin() >= 1)]];
	const size_t n = res.ptr - _view._Unchecked_begin();`

dalle commented on include/fast_float/ascii_number.h:
```diff
@@ -282,38 +286,39 @@ report_parse_error(UC const *p, parse_error error) {
   answer.valid = false;
   answer.too_many_digits = false;
-  // assume p < pend, so dereference without checks;
+  [[assume(p < pend)]]; // assume p < pend, so dereference without checks;
```
Assume C++11 compatibility

IRainman replied:
Hmm, maybe I don't understand correctly, but it's standard since C++23 https://en.cppreference.com/w/cpp/language/attributes/assume and older versions is produced a warning. I need to place this construction into macros, that depends on the platform?

dalle replied:
Yes, the assume attribute is C++23, but this library supports C++11, so that attribute cannot be used.

dalle commented on include/fast_float/float_common.h:
```diff
@@ -63,16 +67,24 @@ template <typename UC> struct from_chars_result_t {
+  const chars_format format
+      // adjust for deprecated feature macros
+#ifdef FASTFLOAT_ALLOWS_LEADING_PLUS
+                            | chars_format::allow_leading_plus
```
This won't compile, and even if it did, it will be overridden by `format(fmt)` in the constructor list.

IRainman replied:
Fix for this https://github.com/fastfloat/fast_float/pull/307/commits/f496321570c6ad7e12dd309934217b6eb3c30805

dalle commented:
I'm sorry to interrupt you, but I think you're going the wrong way here and focusing on the wrong stuff. I don't want to seem pompous or ignorant, and I don't want to offend you, I just don't want you to (possibly) waste more time on small changes.

I will say it clearly: Without proof of the gains there is very low chances that PR will be merged.

So please focus on the tests, make the performance/benchmarks to compile and run. Then you should run the performance/benchmarks workflows. If there are quantified improvements then the likelihood of merging this PR will increase substantially.

lemire commented:
I agree with @dalle.

It is fine to fork the library for your own purposes… it is open source. You can adapt it to your own needs. Please do so.

But we won’t take this PR unless we are convinced of its benefits. And you have not provided the evidence.

We are concerned about maintaining the code, keeping it correct and well tested. 

Changes must be motivated.

IRainman commented:
I added an improvement test with parser emulation.

P. S. In real code of mathematical processing engine, my patches also give significant improvements: 

std::from_chars

Tests: exactly: 115, almost: 9,
 failed: 4,
 time is: 32064ms.

fast_float::from_chars

Tests:
 time is: 26684ms.

( commits from this PR )

Tests (100% of CPU cores is used by BOINC):
 time is: 43135ms.
Tests (50% of CPU cores is used by BOINC):
 time is: 28898ms.
Tests (BOINC is stopped):
 time is: 25459ms.

When switch FASTFLOAT_ONLY_POSITIVE_C_NUMBER_WO_INF_NAN
is applied:

Tests: exactly: 115, almost: 9,
 failed: 4,
 time is: 25947ms.

( commits from this PR )

Tests (100% of CPU cores is used by BOINC):
 time is: 39588ms.
Tests (50% of CPU cores is used by BOINC):
 time is: 27301ms.
Tests (BOINC is stopped):
 time is: 24597ms.

IRainman commented:
Additionally, you can also test the compilation flag FASTFLOAT_ONLY_POSITIVE_C_NUMBER_WO_INF_NAN with a file from here https://population.un.org/wpp/downloads?folder=Standard%20Projections&group=CSV%20format — this is a CSV with only positive numbers in there.

lemire commented:
@IRainman We already have a dataset made entirely of positive values (mesh), see https://github.com/fastfloat/fast_float/pull/308

If you prepare a file that contains just one number per line, you can then pass it to our benchmark like so...
```
cmake -B build -D FASTFLOAT_BENCHMARKS=ON
cmake --build build
./build/benchmarks/realbenchmark myfile.txt
```

(This assumes Linux/macOS... The instructions are similar under Visual Studio except that you need to specify that you work in release mode.)

IRainman commented:
Hmm, I tried to update the brunch, and it's successfully updated, but now this PR isn't updated. What do I need to do? Close this and create a new PR after I finish changes, or may be some command for this situation exists?

lemire commented:
> Hmm, I tried to update the brunch, and it's successfully updated, but now this PR isn't updated. What do I need to do? 

I am not certain of what you mean. The content of a PR is determined by the content of a branch.

IRainman commented:
> I am not certain of what you mean. The content of a PR a determined by the content of a branch.

Now it's probably fine, thank you. I see my new commit in this PR after.

IRainman commented:
Well, I don't know how to compile basictest.cpp locally, and currently I'm done because it's not any ability to compile tests locally on Windows. If you can upgrade the tests I will try again.

IRainman commented:
Now I have added additional improvements, fixed a warning, and made tests in Linux work again.

lemire commented:
@IRainman We cannot merge this as-is. Please break down your PR into smaller contributions, starting from the less consequential.

IRainman commented on include/fast_float/ascii_number.h:
```diff
@@ -350,46 +363,50 @@ parse_number_string(UC const *p, UC const *pend,
-  if ((uint64_t(fmt & chars_format::scientific) && (p != pend) &&
-       ((UC('e') == *p) || (UC('E') == *p))) ||
-      (uint64_t(fmt & detail::basic_fortran_fmt) && (p != pend) &&
-       ((UC('+') == *p) || (UC('-') == *p) || (UC('d') == *p) ||
```
@lemire @dalle Can you confirm that code is correct? I didn't thoroughly understand logic here.

IRainman commented:
![image](https://github.com/user-attachments/assets/621145ff-961d-45d4-8dc4-65214c0a6b83)

IRainman commented:
I don't understand why clang-format is still failing after I formatted all files. What's wrong? 🤷🏻‍♀️ [script for Windows](https://github.com/IRainman/fast_float/blob/main/format.cmd) it's definitely doing that needed.

IRainman commented:
This job is always failed, but clang-format locally confirmed that is already formatted https://github.com/IRainman/fast_float/actions/runs/14861477644/job/41727207426 I don't understand this it all. This is weird.

IRainman committed:
fix for the parse_number_string

IRainman committed:
fix for the parse_number_string

IRainman committed:
fix for the parse_number_string

IRainman committed:
fix for the parse_number_string

IRainman committed:
fix for the parse_number_string

IRainman commented:
I'm probably done, you can review and do some speed tests. Now all tests are working. Can you please give me advice, what test I need to add for my changes?

IRainman committed:
additional fix for bfloat16_t. Sorry, I can't compile it's locally.

IRainman committed:
additional FASTFLOAT_HAS_BIT_CAST improve for older standards.

IRainman committed:
fix for type usage in parse_int_string

IRainman committed:
cleanup for parse_number_string

IRainman committed:
compilation fix for internal tests

IRainman committed:
unfck clang format

IRainman committed:
Merge branch 'main' of https://github.com/fastfloat/fast_float

IRainman commented:
[original.txt](https://github.com/user-attachments/files/20741045/original.txt)
[pr307_with_flags.txt](https://github.com/user-attachments/files/20741046/pr307_with_flags.txt)
[pr307.txt](https://github.com/user-attachments/files/20741047/pr307.txt)

IRainman commented:
@lemire please refer to the numbers in the comment above this.

dependabot[bot] committed:
Bump actions/checkout from 4 to 5 in the github-actions group

Bumps the github-actions group with 1 update: [actions/checkout](https://github.com/actions/checkout).


Updates `actions/checkout` from 4 to 5
- [Release notes](https://github.com/actions/checkout/releases)
- [Commits](https://github.com/actions/checkout/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

IRainman committed:
Merge branch 'fastfloat:main' into main

IRainman committed:
Merge pull request #2 from IRainman/dependabot/github_actions/github-actions-a331d3ec2d

Bump actions/checkout from 4 to 5 in the github-actions group

IRainman committed:
Merge branch 'main' of https://github.com/fastfloat/fast_float

IRainman committed:
# fix for stackvec constructor.

IRainman committed:
# cleanup

IRainman committed:
Merge branch 'main' of https://github.com/fastfloat/fast_float

IRainman committed:
* Readme updated.

IRainman committed:
# compilation fix after merge from upstream.

IRainman reviewed (commented):
Upgraded to 8.1.0

IRainman committed:
# clang format

IRainman committed:
# types fix after merge from upstream.

IRainman committed:
Update README.md

dependabot[bot] committed:
Bump the github-actions group across 1 directory with 3 updates

Bumps the github-actions group with 3 updates in the / directory: [actions/checkout](https://github.com/actions/checkout), [github/codeql-action](https://github.com/github/codeql-action) and [actions/setup-node](https://github.com/actions/setup-node).


Updates `actions/checkout` from 4 to 5
- [Release notes](https://github.com/actions/checkout/releases)
- [Commits](https://github.com/actions/checkout/compare/v4...v5)

Updates `github/codeql-action` from 3 to 4
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v3...v4)

Updates `actions/setup-node` from 4.4.0 to 5.0.0
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/49933ea5288caeca8642d1e84afbd3f7d6820020...a0853c24544627f65ddf259abe73b1d18a591444)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
- dependency-name: github/codeql-action
  dependency-version: '4'
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
- dependency-name: actions/setup-node
  dependency-version: 5.0.0
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

IRainman committed:
Merge branch 'main' of https://github.com/fastfloat/fast_float

IRainman committed:
# compilation fix after merge.

IRainman committed:
# format

IRainman committed:
Merge pull request #5 from IRainman/dependabot/github_actions/github-actions-28186ec776

Bump the github-actions group across 1 directory with 3 updates

IRainman committed:
# compilation fix after merge

IRainman committed:
Merge branch 'main' of https://github.com/IRainman/fast_float

IRainman committed:
# tests are updated

IRainman committed:
# improve checking for integer in the FASTFLOAT_ONLY_POSITIVE_C_NUMBER_WO_INF_NAN mode.

IRainman committed:
# const and constexpr fixes
# types fixes.
# FASTFLOAT_ASSERT fix.

IRainman committed:
# minimize diff in the crutch.

IRainman committed:
# fix for digit_comp.

IRainman committed:
# cleanup and speedup digit_comp.

IRainman committed:
# cleanup.

IRainman committed:
# types fix.

IRainman committed:
# type usage fix.

IRainman committed:
* Upgrade manually vectorized code to SSE4.2 for FASTFLOAT_64BIT because it's already used SSE4.1 instructions.

IRainman committed:
* Upgrade manually vectorized code to support SSE4.2.

IRainman committed:
* Upgrade manually vectorized code to support SSE4.2, final cleanup and documentation for SSE2 code.

IRainman committed:
# fix for FASTFLOAT_64BIT

IRainman committed:
# cleanup

IRainman committed:
* fix for parse_eight_digits_unrolled

IRainman committed:
# format

IRainman committed:
# small cleanup

dependabot[bot] committed:
Bump the github-actions group across 1 directory with 3 updates

Bumps the github-actions group with 3 updates in the / directory: [actions/upload-artifact](https://github.com/actions/upload-artifact), [actions/setup-node](https://github.com/actions/setup-node) and [jidicula/clang-format-action](https://github.com/jidicula/clang-format-action).


Updates `actions/upload-artifact` from 4 to 5
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v4...v5)

Updates `actions/setup-node` from 5.0.0 to 6.0.0
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/a0853c24544627f65ddf259abe73b1d18a591444...2028fbc5c25fe9cf00d9f06a71cc4710d4507903)

Updates `jidicula/clang-format-action` from 4.15.0 to 4.16.0
- [Release notes](https://github.com/jidicula/clang-format-action/releases)
- [Commits](https://github.com/jidicula/clang-format-action/compare/4726374d1aa3c6aecf132e5197e498979588ebc8...6cd220de46c89139a0365edae93eee8eb30ca8fe)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
- dependency-name: actions/setup-node
  dependency-version: 6.0.0
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
- dependency-name: jidicula/clang-format-action
  dependency-version: 4.16.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

IRainman committed:
Merge pull request #7 from IRainman/dependabot/github_actions/github-actions-c0f80da9d0

Bump the github-actions group across 1 directory with 3 updates

IRainman committed:
Merge branch 'main' of https://github.com/fastfloat/fast_float

IRainman committed:
Merge branch 'main' of https://github.com/IRainman/fast_float

IRainman committed:
Merge branch 'fastfloat:main' into main

IRainman committed:
Merge branch 'main' of https://github.com/IRainman/fast_float

IRainman committed:
* Improving in code generation -> speedup.
* fix warning in 32 bit build.

IRainman committed:
# Disable FASTFLOAT_HAS_BYTESWAP: step to try to fix precision error on x86 platform

IRainman committed:
# test failed.

IRainman committed:
* try to fix precision error on x86 platform step2.

IRainman committed:
* try to fix precision error on x86 platform step3.

IRainman committed:
* try to fix precision error on x86 platform step4.

IRainman committed:
* try to fix precision error on x86 platform step5.

IRainman committed:
* try to fix precision error on x86 platform step6.

IRainman committed:
* try to fix precision error on x86 platform step7.

IRainman committed:
* try to fix precision error on x86 platform step8.

IRainman committed:
* try to fix precision error on x86 platform step9.

IRainman committed:
* type usage fix

IRainman committed:
# tests

IRainman committed:
* try to fix error on x86 platform step1.

IRainman committed:
* code cleanup.

IRainman committed:
* try to fix error on x86 platform step2.

IRainman committed:
# unfck lint

IRainman committed:
* try to fix error on x86 platform step3.

IRainman committed:
* code cleanup.

IRainman committed:
# cycles (for and while) cleanup in low level for the best compiler optimization and the best runtime.

IRainman committed:
# unfck lint

IRainman committed:
* Added separate config macros FASTFLOAT_ISNOT_CHECKED_BOUNDS.

IRainman reviewed (commented):
Needs custom check.

IRainman committed:
* type usage fix for better performance in any hardware.

IRainman committed:
# test fixed.

IRainman committed:
# cleanup.

IRainman committed:
* improvements in compiler function detection.

IRainman committed:
* FASTFLOAT_HAS_BIT_CAST cleanup.

IRainman committed:
# warning fix.

IRainman committed:
Merge branch 'main' of https://github.com/fastfloat/fast_float

dependabot[bot] committed:
Bump the github-actions group across 1 directory with 2 updates

Bumps the github-actions group with 2 updates in the / directory: [actions/checkout](https://github.com/actions/checkout) and [actions/setup-node](https://github.com/actions/setup-node).


Updates `actions/checkout` from 5 to 6
- [Release notes](https://github.com/actions/checkout/releases)
- [Commits](https://github.com/actions/checkout/compare/v5...v6)

Updates `actions/setup-node` from 6.0.0 to 6.1.0
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/2028fbc5c25fe9cf00d9f06a71cc4710d4507903...395ad3262231945c25e8478fd5baf05154b1d79f)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
- dependency-name: actions/setup-node
  dependency-version: 6.1.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

IRainman committed:
Merge branch 'fastfloat:main' into main

IRainman committed:
Merge branch 'main' of https://github.com/IRainman/fast_float

IRainman committed:
Merge pull request #9 from IRainman/dependabot/github_actions/github-actions-5c52a8f22e

Bump the github-actions group across 1 directory with 2 updates

IRainman committed:
* type usage fix for better performance in any hardware.

IRainman committed:
Merge branch 'main' of https://github.com/IRainman/fast_float

IRainman committed:
Benchmark are updated.

dependabot[bot] committed:
Bump the github-actions group with 2 updates

Bumps the github-actions group with 2 updates: [actions/checkout](https://github.com/actions/checkout) and [actions/upload-artifact](https://github.com/actions/upload-artifact).


Updates `actions/checkout` from 5.0.1 to 6.0.1
- [Release notes](https://github.com/actions/checkout/releases)
- [Commits](https://github.com/actions/checkout/compare/v5.0.1...v6.0.1)

Updates `actions/upload-artifact` from 5 to 6
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v5...v6)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: 6.0.1
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
- dependency-name: actions/upload-artifact
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

IRainman committed:
Merge branch 'main' of https://github.com/fastfloat/fast_float

IRainman committed:
Merge pull request #10 from IRainman/dependabot/github_actions/github-actions-298b600d03

Bump the github-actions group with 2 updates

IRainman committed:
* enable warning in GCC because PVS-Studio are also detect this. This is an error!

IRainman committed:
Merge branch 'main' of https://github.com/IRainman/fast_float

IRainman committed:
* enable warning in GCC because PVS-Studio are also detect this. This is an error!

IRainman committed:
* Small optimization in code generation for auto vectorization.

IRainman committed:
Merge branch 'main' of https://github.com/fastfloat/fast_float

IRainman committed:
* Added additional compile option FASTFLOAT_TABLE_HACK_CHAR_DIGIT_LUT_DISABLED for improve cache usage in high load.
* Small optimization in code generation for auto vectorization.

IRainman committed:
* optimize layout of the parsed_number_string_t.

IRainman committed:
unfck lint.

IRainman committed:
FASTFLOAT_HAS_BYTESWAP cleanup.

IRainman committed:
MSVC compilation fix.

IRainman committed:
small fix.

IRainman committed:
small fix

IRainman committed:
fix for biodegradable code in doctest.

IRainman committed:
# constexpr noexcept

IRainman committed:
Final review for a more simple merge.
Feature FASTFLOAT_TABLE_HACK_CHAR_DIGIT_LUT_DISABLED is removed because hex and bin parsing isn't working properly.
Properly use FASTFLOAT_SIMD_DISABLE_WARNINGS and FASTFLOAT_SIMD_RESTORE_WARNINGS only for instructions that allow unaligned loads.

IRainman committed:
unfck lint

IRainman committed:
revert workflows.

IRainman committed:
Doctest DOCTEST_CONFIG_SUPER_FAST_ASSERTS

IRainman committed:
fucking lint

IRainman committed:
Update the Readme with fmt to the high performance example.

IRainman committed:
Enhance README with compiler optimization details

Updated performance recommendations and compiler flags for C++.

IRainman committed:
am_pow_t is probably needs to be 64 bit because some hard coded values that I can't find.

IRainman committed:
try again. need test refactoring.

IRainman committed:
remove format.cmd: use IDE.

IRainman committed:
* documentation in the code updated, fixed errors with parsing some special very big exponent numbers.

IRainman committed:
fix compilation error in uint8_t parsing.

IRainman committed:
code cleanup and type usage fixes.

IRainman committed:
type usage fixes.

IRainman reviewed (commented):
Well, today I fixed the last error in the parsing; tomorrow I'll test error code processing in multibyte strings because it's the last problem.

IRainman committed:
Merge branch 'main' of https://github.com/fastfloat/fast_float

IRainman committed:
type usage fixes.

IRainman committed:
cleanup.

IRainman committed:
type usage fixes.
fix for FASTFLOAT_ISNOT_CHECKED_BOUNDS if FASTFLOAT_ONLY_POSITIVE_C_NUMBER_WO_INF_NAN isn't enabled.

IRainman committed:
type usage fixes.

IRainman committed:
noexcept

IRainman committed:
type usage fixes.

IRainman committed:
type usage fixes.

IRainman committed:
cleanup and type usage fixes.

IRainman committed:
optimization for the report_parse_error.

IRainman committed:
initialization cleanup and improve caches usage.

IRainman committed:
type usage fix.

IRainman committed:
some warnings fix in CI builds.

IRainman committed:
initialization cleanup.

IRainman committed:
initialization cleanup.

IRainman committed:
type usage fix and cleanup.

IRainman committed:
type usage fixes.

IRainman committed:
type usage fixes.

IRainman committed:
type usage fixes.

IRainman committed:
type usage fixes.

IRainman committed:
type usage fixes.

IRainman committed:
type usage fix.

IRainman committed:
cleanup.

IRainman committed:
cleanup.

IRainman committed:
interface cleanup.

IRainman committed:
* small cleanup in the tests/basictest.cpp

IRainman committed:
type usage fixes.

IRainman commented:
<img width="2682" height="989" alt="image" src="https://github.com/user-attachments/assets/fede1f0d-1cc8-4130-9bf4-843c7b76a14f" />
@dalle @lemire updated performance comparison.

IRainman committed:
Merge branch 'main' of https://github.com/fastfloat/fast_float

IRainman committed:
cleanup

IRainman committed:
small performance improvement after full cleanup.

IRainman committed:
FASTFLOAT_HAS_BIT_CAST fix for old standards.

IRainman committed:
cleanup in the API.

IRainman committed:
type usage fix.

IRainman committed:
type usage fix.

IRainman committed:
added compilation flags to the bench ip.

IRainman committed:
improvements in the code generation for byteswap and leading_zero to all compilers.

IRainman committed:
byteswap isn't compiling properly, no nay crutches that I try not help, including usage cpp20_and_in_constexpr.

IRainman committed:
small update for benchmark.

IRainman committed:
compilation fix.

IRainman committed:
Merge branch 'main' of https://github.com/fastfloat/fast_float

IRainman committed:
copy-past fix and type usage fixes.

IRainman committed:
Fixes for FASTFLOAT_IS_BIG_ENDIAN after PR #359 is added to original.

IRainman committed:
cleanup and fix for uint8_t and uint16_t parsing.

IRainman committed:
fix for integer_times_pow10.

IRainman committed:
Fix for parse_int_string, not all compilers work properly.

IRainman committed:
nope, my implementation is fine. Needs compilers and tests check:

 WARNING: The requested image's platform (linux/s390x) does not match the detected host platform (linux/amd64/v3) and no specific platform was requested

IRainman committed:
try to cleanup crutches: it's not needed anymore?

IRainman committed:
workaround needs for both gcc and clang.

IRainman committed:
fix for parse_int_string.

IRainman committed:
cleanup. all others bug in the compilers or in the workflows.

IRainman committed:
cleanup.

IRainman reviewed (commented):
Currently I'm done. All other errors are probably from compilers or from workflows.

IRainman committed:
Update CI badge and add CodeFactor badge

IRainman committed:
Update README badges for Ubuntu 24.04

IRainman committed:
Merge branch 'main' of https://github.com/fastfloat/fast_float

IRainman committed:
Merge branch 'main' of https://github.com/fastfloat/fast_float

dependabot[bot] committed:
Bump the github-actions group across 1 directory with 2 updates

Bumps the github-actions group with 2 updates in the / directory: [actions/checkout](https://github.com/actions/checkout) and [actions/setup-node](https://github.com/actions/setup-node).


Updates `actions/checkout` from 5.0.1 to 6.0.1
- [Release notes](https://github.com/actions/checkout/releases)
- [Commits](https://github.com/actions/checkout/compare/v5.0.1...v6.0.1)

Updates `actions/setup-node` from 6.1.0 to 6.2.0
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/395ad3262231945c25e8478fd5baf05154b1d79f...6044e13b5dc448c55e2357c09f80417699197238)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: 6.0.1
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
- dependency-name: actions/setup-node
  dependency-version: 6.2.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

IRainman committed:
Merge branch 'main' of https://github.com/IRainman/fast_float

IRainman committed:
Cleanup after merge.

IRainman committed:
unfck lint.

IRainman committed:
Merge pull request #13 from IRainman/dependabot/github_actions/github-actions-c132dce648

Bump the github-actions group across 1 directory with 2 updates

IRainman committed:
harder code optimization ;) also in the test that show the error.

IRainman committed:
Merge branch 'main' of https://github.com/IRainman/fast_float

IRainman committed:
cleanup test for the best compilation and more aggressive optimization to test errors in the code.

IRainman committed:
cleanup, unfortunately, the patch isn't correct. fastfloat_strncasecmp3 fastfloat_strncasecmp5 and fastfloat_strncasecmp needs to be revised by the author.

IRainman committed:
review of the fastfloat_strncasecmp3 fastfloat_strncasecmp5 fastfloat_strncasecmp and parse_infnan

IRainman committed:
gcc build fix

IRainman committed:
include fix.

IRainman committed:
cleanup.

IRainman committed:
fix for old C++ standard in test.

IRainman committed:
fix for old C++ standard in library.

IRainman committed:
fix for old standard.

IRainman committed:
Add CI badges for various platforms in README

IRainman committed:
fix for gcc.

IRainman committed:
Merge branch 'main' of https://github.com/IRainman/fast_float

IRainman committed:
style cleanup.

IRainman committed:
update for copy-pasted tests. meh. needs cleanup.

IRainman committed:
unfck lint

IRainman committed:
isolate clang crutch.

IRainman committed:
unfck lint

IRainman committed:
tests cleanup

IRainman committed:
Merge branch 'fastfloat:main' into main

dependabot[bot] committed:
Bump actions/checkout from 6.0.1 to 6.0.2 in the github-actions group

Bumps the github-actions group with 1 update: [actions/checkout](https://github.com/actions/checkout).


Updates `actions/checkout` from 6.0.1 to 6.0.2
- [Release notes](https://github.com/actions/checkout/releases)
- [Commits](https://github.com/actions/checkout/compare/v6.0.1...v6.0.2)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: 6.0.2
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

IRainman committed:
Merge branch 'main' of https://github.com/fastfloat/fast_float

IRainman committed:
Merge pull request #14 from IRainman/dependabot/github_actions/github-actions-e6ee9d7de3

Bump actions/checkout from 6.0.1 to 6.0.2 in the github-actions group

IRainman committed:
Merge branch 'fastfloat:main' into main

IRainman committed:
Merge branch 'main' of https://github.com/fastfloat/fast_float

dependabot[bot] committed:
Bump jidicula/clang-format-action

Bumps the github-actions group with 1 update in the / directory: [jidicula/clang-format-action](https://github.com/jidicula/clang-format-action).


Updates `jidicula/clang-format-action` from 4.17.0 to 4.18.0
- [Release notes](https://github.com/jidicula/clang-format-action/releases)
- [Commits](https://github.com/jidicula/clang-format-action/compare/3a18028048f01a66653b4e3bf8d968d2e7e2cb8b...654a770daa28443dd111d133e4083e21c1075674)

---
updated-dependencies:
- dependency-name: jidicula/clang-format-action
  dependency-version: 4.18.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

IRainman committed:
Compilation fix for Ubuntu 24.04, updated docs.

IRainman committed:
Merge pull request #18 from IRainman/dependabot/github_actions/github-actions-1f9dd65b39

Bump jidicula/clang-format-action from 4.17.0 to 4.18.0 in the github-actions group across 1 directory

dependabot[bot] committed:
Bump mymindstorm/setup-emsdk from 14 to 15 in the github-actions group

Bumps the github-actions group with 1 update: [mymindstorm/setup-emsdk](https://github.com/mymindstorm/setup-emsdk).


Updates `mymindstorm/setup-emsdk` from 14 to 15
- [Release notes](https://github.com/mymindstorm/setup-emsdk/releases)
- [Commits](https://github.com/mymindstorm/setup-emsdk/compare/6ab9eb1bda2574c4ddb79809fc9247783eaf9021...667eb33f24e84e7f362c16d8d7fff0629a73e15e)

---
updated-dependencies:
- dependency-name: mymindstorm/setup-emsdk
  dependency-version: '15'
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

IRainman committed:
Merge branch 'main' of https://github.com/fastfloat/fast_float

IRainman committed:
Add VS18 workflows

IRainman committed:
Merge pull request #19 from IRainman/dependabot/github_actions/github-actions-833d03c70b

Bump mymindstorm/setup-emsdk from 14 to 15 in the github-actions group

IRainman committed:
Update README.md

IRainman committed:
Update README.md

IRainman committed:
Update README.md

IRainman committed:
Update README.md

IRainman committed:
Update README.md

dependabot[bot] committed:
Bump the github-actions group across 1 directory with 2 updates

Bumps the github-actions group with 2 updates in the / directory: [actions/setup-node](https://github.com/actions/setup-node) and [mymindstorm/setup-emsdk](https://github.com/mymindstorm/setup-emsdk).


Updates `actions/setup-node` from 6.3.0 to 6.4.0
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/53b83947a5a98c8d113130e565377fae1a50d02f...48b55a011bda9f5d6aeb4c2d9c7362e8dae4041e)

Updates `mymindstorm/setup-emsdk` from 15 to 16
- [Release notes](https://github.com/mymindstorm/setup-emsdk/releases)
- [Commits](https://github.com/mymindstorm/setup-emsdk/compare/667eb33f24e84e7f362c16d8d7fff0629a73e15e...4528d102f7230f0e7b276855c01ea1159be0e984)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-version: 6.4.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: github-actions
- dependency-name: mymindstorm/setup-emsdk
  dependency-version: '16'
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

IRainman committed:
Merge branch 'main' of https://github.com/fastfloat/fast_float

IRainman committed:
Merge pull request #21 from IRainman/dependabot/github_actions/github-actions-100cea0b56

Bump the github-actions group across 1 directory with 2 updates

IRainman committed:
HedgehogInTheCPP 8.2.5 version.

IRainman committed:
Merge branch 'main' of https://github.com/IRainman/fast_float

IRainman committed:
Added PVS-Studio.

IRainman committed:
Fixed issues found by PVS-Studio

IRainman committed:
small performance fix.

IRainman committed:
cleanup.

IRainman commented:
Progress for today:
<img width="2723" height="1162" alt="image" src="https://github.com/user-attachments/assets/9abe7791-d1c2-4d32-a70e-899077158edc" />

IRainman committed:
Merge branch 'main' of https://github.com/fastfloat/fast_float

IRainman committed:
# alignment next step performance: original now perform my program test for 133704ms and my fork give more significant performance improve than before: 105259ms.

IRainman committed:
* type usage fix in merged code

IRainman committed:
* const fixes for merged code.

dependabot[bot] committed:
Bump actions/checkout in the github-actions group across 1 directory

Bumps the github-actions group with 1 update in the / directory: [actions/checkout](https://github.com/actions/checkout).


Updates `actions/checkout` from 6 to 7
- [Release notes](https://github.com/actions/checkout/releases)
- [Commits](https://github.com/actions/checkout/compare/v6...v7)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '7'
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

IRainman committed:
Merge pull request #22 from IRainman/dependabot/github_actions/github-actions-640176b5ab

Bump actions/checkout from 6 to 7 in the github-actions group across 1 directory

IRainman committed:
* copy-past cleanup.

IRainman committed:
Merge branch 'main' of https://github.com/IRainman/fast_float

IRainman committed:
# cleanup

IRainman committed:
# compilation fix after renaming

IRainman committed:
# cleanup

IRainman committed:
# cleanup

IRainman committed:
* added crutch for compilers that do not properly support constexpr if

IRainman committed:
# lin
t unfck

IRainman committed:
Add CI badges for Visual Studio 2018 workflows

IRainman committed:
* cleanup and optimize more.

IRainman committed:
Merge branch 'main' of https://github.com/IRainman/fast_float

IRainman committed:
* cleanup and slightly optimize slower path.

IRainman committed:
* signed/unsigned mismatch fix.

IRainman committed:
Merge branch 'main' of https://github.com/fastfloat/fast_float

IRainman committed:
lindt unfck

IRainman committed:
* small cleanup for the long path
